### ISSUE TITLE
"Zombio bio-operator is unaffected by EMP Blast" despite no vision and on different Z level

### ISSUE THREAD

**Issue by CoroNaut** [CleverRaven/Cataclysm-DDA#36486]
# Describe the bug
I am getting a message that requires line of sight for an event that is not in line of sight.
# Steps To Reproduce

1. Have bio-operator being affected by an emp blast on some other Z-level but near you ofcourse.
2. get the message 

# Expected behavior
Can't see this message if not in line of sight

# Screenshots
![Capture](https://user-images.githubusercontent.com/24307650/71504193-7ab93000-2846-11ea-93c2-226bf6aafdc5.PNG)

# Versions and configuration
- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-10211-g0bd1ad2 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    No Fungal Monsters [No_Fungi],
    Disable Religious Texts [no_religious_Texts]
]

**Comment by ZhilkinSerg** [comment]
This message is guarded by `sight` (which is `const bool sight = g->u.sees( p )`), so it should be a bug of vision.

**Comment by CoroNaut** [comment]
I also had another very very strange bug,  a turret was firing into a wall, and there was nothing behind that wall and wouldn't stop until i examined and removed it, I thought maybe it was shooting to a different z-level, but there was definitely no breaks in floor or ceiling. This bug and the posted bug may have been a weird edge case with the experimental z-levels.

**Comment by Fris0uman** [comment]
I can only reproduce by dropping and EMP in a crowd of bio op and teleporting above before it blows  up. And code says that EMP blast don't actually travel accros z-level
 ```cpp
void emp_blast( const tripoint &p )
{
    // TODO: Implement z part
    int x = p.x;
    int y = p.y;
```
This issue of getting a message for something you can't see anymore  might be related to #36820

**Comment by CoroNaut** [comment]
I went back to the location with a debug clairvoyance artifact, I couldn't find any zombie bio-operators dead or alive from when i ransacked the place in any situation that could've produced the message. I even spawned a bunch of electricity-generating zombies and even the the shock vent field and couldn't get the message to pop up. I didn't use any emp grenades and hadn't teleported when initially going through. That lab in the picture is also extremely massive, spanning from Z-9 to Z-2 with upwards of 450 lab map-tiles

My best guess is that there is something causing the game to just desync or make funny things happen after playing a long time. When I play, I usually keep the game open and go back and forth between it and other tasks. This ends up making the game open and being played for maybe 10 hours sometimes. Once again, just a complete guess.

Anyways, I'm gonna close this issue, even though I encountered this sight bug with the zombie bio-operator and the turret. I can't reproduce and I should've put up a save, my world is approaching 600MB, so I'm not sure anyone wanted that beast of a save file for such an edge case bug lol.
